src/python/pants/engine/target.py

```diff
@@ -1004,28 +972,14 @@ def __init__(self, generator: Target, generated_targets: Iterable[Target]) -> No
         super().__init__(mapping)
 
 
-class TargetTypesToGenerateTargetsRequests(FrozenDict[Type[Target], Type[GenerateTargetsRequest]]):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        for tgt_cls, request in self.items():
-            # TODO: After this deprecation, the key type for this class should become
-            # `TargetGenerator`, and the `request_for` method should require a `TargetGenerator`
-            # argument, so that callers are forced to cast / check subtyping before calling.
-            deprecated_conditional(
-                lambda: not issubclass(tgt_cls, TargetGenerator),
-                removal_version="2.12.0.dev2",
-                entity="`GenerateTargetsRequest.generate_from` types which do not subclass `TargetGenerator`",
-                hint=(
-                    f"The generate_from type of `{request.__name__}` was `{tgt_cls.__name__}`, "
-                    "which does not subclass `TargetGenerator`."
-                ),
-            )
-
-    def is_generator(self, tgt: Target) -> bool:
+class TargetTypesToGenerateTargetsRequests(
+    FrozenDict[Type[TargetGenerator], Type[GenerateTargetsRequest]]
+):
+    def is_generator(self, tgt: TargetGenerator) -> bool:
```

Comment:
Nit: based on the callsites, it would maybe be a handy convenience for `is_generator` to continue to take `Target` and do the cast internally, since it looks like callers of `request_for` are the ones who should be forced to do it manually.

Reply:
Yeah - fixed.